src/app-layout/visual-refresh-toolbar/notifications/styles.scss

```diff
@@ -2,7 +2,11 @@
  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  SPDX-License-Identifier: Apache-2.0
 */
-.sticky-notifications {
-  position: sticky;
-  z-index: 850;
+@use '../../../internal/styles/' as styles;
+
+@include styles.media-breakpoint-up(styles.$breakpoint-x-small) {
```

Comment:
I already have a fix for this issue in my draft PR: https://github.com/cloudscape-design/components/pull/2761/files?w=1#diff-3cfa742d60cd325bf84e2694ddf9a04bf3ebdda6cd27919a6dec23caac3953d0R178

Ack on your PR, I will consolidate it into mine and merge all together